lib/Template.pm

```diff
@@ -192,6 +192,9 @@ sub _output {
         elsif (open(FP, '>', $where)) {
             # binmode option can be 1 or a specific layer, e.g. :utf8
             my $bm = $options->{ binmode  };
+            if (not(defined $bm)) {
```

Comment:
style comment: not sure why you are using `not` instead of `!` and why you are using some extra parens there
otherwise this looks good